server/channels/app/status_test.go

```diff
@@ -177,7 +177,11 @@ func TestSetCustomStatus(t *testing.T) {
 	} {
 		t.Run(testCase.Name, func(t *testing.T) {
 			err := th.App.SetCustomStatus(th.Context, th.BasicUser.Id, testCase.Input)
-			defer th.App.RemoveCustomStatus(th.Context, th.BasicUser.Id)
+			require.Nil(t, err)
+			defer func() {
+				err = th.App.RemoveCustomStatus(th.Context, th.BasicUser.Id)
+				require.Nil(t, err)
+			}()
```

Comment:
The test is failing with 

```
        	Error:      	Expected nil, but got: &model.AppError{Id:"api.custom_status.set_custom_statuses.emoji_not_found", Message:"Failed to update the custom status. An emoji with the given name does not exist.", DetailedError:"", RequestId:"", StatusCode:400, Where:"SetCustomStatus", SkipTranslation:false, params:map[string]interface {}(nil), wrapped:(*model.AppError)(0xc0060f8600)}

```

I don't think you need to call `RemoveCustomStatus` in every case. If the call to `SetCustomStatus` failed, there is no need for the `defer` cleanup.

Reply:
Roger